File: spring-boot-iot-system/src/main/java/com/ghlzm/iot/system/entity/Dict.java
package com.ghlzm.iot.system.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.io.Serializable;
import java.util.Date;
import java.util.List;

/**
 * 字典配置实体
 */
@Data
@TableName("sys_dict")
public class Dict implements Serializable {
      private static final long serialVersionUID = 1L;

      /**
       * 主键
       */
      @TableId(value = "id", type = IdType.ASSIGN_ID)
      private Long id;

      /**
       * 租户ID
       */
      private Long tenantId;

      /**
       * 字典名称
       */
      private String dictName;

      /**
       * 字典编码
       */
      private String dictCode;

      /**
       * 字典类型 text/number/boolean/date
       */
      private String dictType;

      /**
       * 状态 1启用 0禁用
       */
      private Integer status;

      /**
       * 排序
       */
      private Integer sortNo;

      /**
       * 备注
       */
      private String remark;

      /**
       * 创建人
       */
      private Long createBy;

      /**
       * 创建时间
       */
      private Date createTime;

      /**
       * 更新人
       */
      private Long updateBy;

      /**
       * 更新时间
       */
      private Date updateTime;

      /**
       * 删除标记
       */
      @TableLogic
      private Integer deleted;

      /**
       * 字典项列表
       */
      @TableField(exist = false)
      private List<DictItem> items;
}

File: spring-boot-iot-ui/src/api/dict.ts
import { request } from './request'
import type { ApiEnvelope, IdType, PageResult } from '../types/api'

export interface DictItem {
  id: IdType
  tenantId: IdType
  dictId: IdType
  itemName: string
  itemValue: string
  itemType: string
  status: number
  sortNo: number
  remark: string
  createBy: number
  createTime: string
  updateBy: number
  updateTime: string
  deleted: number
}

export interface Dict {
  id: IdType
  tenantId: IdType
  dictName: string
  dictCode: string
  dictType: string
  status: number
  sortNo: number
  remark: string
  createBy: number
  createTime: string
  updateBy: number
  updateTime: string
  deleted: number
  items?: DictItem[]
}

export interface DictQueryParams {
  dictName?: string
  dictCode?: string
  dictType?: string
}

export interface DictPageQueryParams extends DictQueryParams {
  pageNum?: number
  pageSize?: number
}

function buildQuery(params: Record<string, unknown>) {
  const query = new URLSearchParams()
  Object.entries(params).forEach(([key, value]) => {
    if (value !== undefined && value !== null && value !== '') {
      query.append(key, String(value))
    }
  })
  return query.toString()
}

export const listDicts = (params: DictQueryParams = {}): Promise<ApiEnvelope<Dict[]>> => {
  const query = buildQuery(params)
  return request<Dict[]>(`/api/dict/list${query ? `?${query}` : ''}`, { method: 'GET' })
}

export const pageDicts = (params: DictPageQueryParams = {}): Promise<ApiEnvelope<PageResult<Dict>>> => {
  const query = buildQuery(params)
  return request<PageResult<Dict>>(`/api/dict/page${query ? `?${query}` : ''}`, { method: 'GET' })
}

export const listDictTree = (): Promise<ApiEnvelope<Dict[]>> => {
  return request<Dict[]>('/api/dict/tree', { method: 'GET' })
}

export const getDict = (id: IdType): Promise<ApiEnvelope<Dict>> => {
  return request<Dict>(`/api/dict/${id}`, { method: 'GET' })
}

export const getDictByCode = (dictCode: string): Promise<ApiEnvelope<Dict>> => {
  return request<Dict>(`/api/dict/code/${dictCode}`, { method: 'GET' })
}

export const addDict = (data: Partial<Dict>): Promise<ApiEnvelope<Dict>> => {
  return request<Dict>('/api/dict', { method: 'POST', body: data })
}

export const updateDict = (data: Partial<Dict>): Promise<ApiEnvelope<Dict>> => {
  return request<Dict>('/api/dict', { method: 'PUT', body: data })
}

export const deleteDict = (id: IdType): Promise<ApiEnvelope<void>> => {
  return request<void>(`/api/dict/${id}`, { method: 'DELETE' })
}

export const listDictItems = (dictId: IdType): Promise<ApiEnvelope<DictItem[]>> => {
  return request<DictItem[]>(`/api/dict/${dictId}/items`, { method: 'GET' })
}

export const addDictItem = (data: Partial<DictItem>): Promise<ApiEnvelope<DictItem>> => {
  return request<DictItem>(`/api/dict/${data.dictId}/items`, { method: 'POST', body: data })
}

export const updateDictItem = (data: Partial<DictItem>): Promise<ApiEnvelope<DictItem>> => {
  return request<DictItem>(`/api/dict/${data.dictId}/items`, { method: 'PUT', body: data })
}

export const deleteDictItem = (id: IdType): Promise<ApiEnvelope<void>> => {
  return request<void>(`/api/dict/items/${id}`, { method: 'DELETE' })
}
